discovery/manager.go

```diff
@@ -106,12 +107,12 @@ type provider struct {
 	config interface{}
 }
 
-// NewManager is the Discovery Manager constructor
-func NewManager(ctx context.Context, logger log.Logger) *Manager {
+// NewManager is the Discovery Manager constructor.
+func NewManager(ctx context.Context, logger log.Logger, options ...func(*Manager)) *Manager {
```

Comment:
I didn't want to change the signatute of `NewManager()` as it is imported by file-based discovery systems. I stole the idea of variadic option functions from https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis